src/fpSingleton/dataset/inputMatrixDataColMajor.h

```diff
@@ -0,0 +1,95 @@
+#ifndef inputMatrixDataColMajor_h
```

Comment:
This seems like a lot of repeated code -- could we merge with inputMatrixData or use a base class that they both inherit from and have custom methods for `returnFeatureValue` and `prefetchFeatureValue`

Reply:
resolved in #11